src/com/company/redi_exercises/Main.java

```diff
@@ -0,0 +1,186 @@
+        firstList.add(1);
+        firstList.add(2);
+        firstList.add(3);
+        firstList.add(4);
+        ArrayList<Integer> secondList = new ArrayList<>();
+        secondList.add(1);
+        secondList.add(4);
+        secondList.add(5);
+        secondList.add(9);
+
+        ArrayList<Integer> earnedList = new ArrayList<>();
+
+        findDistinctNumbers(firstList, secondList, earnedList);
+        findDistinctNumbers(secondList, firstList, earnedList);
+
+        System.out.println("Output");
+        System.out.println(earnedList);
+    }
+
+    private static void findDistinctNumbers(ArrayList<Integer> firstList, ArrayList<Integer> secondList, ArrayList<Integer> earnedList) {
+        for (int number : firstList) {
+            if (!secondList.contains(number)) {
+                earnedList.add(number);
+            }
+        }
+    }
+
+    private static void thirteenthExercise() {
+        ArrayList<String> inputWords = new ArrayList<>();
+        inputWords.add("hello");
+        inputWords.add("world");
+        inputWords.add("hello");
+        inputWords.add("germany");
+        inputWords.add("hello");
+        inputWords.add("berlin");
+
+        HashSet<String> distinctWords = new HashSet<>();
+        distinctWords.addAll(inputWords);
+        System.out.println(distinctWords);
+
+        int count = 0;
+        for (String checkWord : distinctWords) {
+            for (String word : inputWords) {
+                if (checkWord == word) {
+                    count++;
+                }
+            }
+            System.out.println(checkWord + ", count: " + count);
+            count = 0;
+        }
+    }
+
+    private static void thirteenthExercisePartTwo(Scanner scanner) {
+        ArrayList<Integer> inputWords = new ArrayList<>();
+        int newNumber = 0;
+        while (true) {
+            newNumber = scanner.nextInt();
+
+            if (newNumber == -1) {
+                break;
+            }
+            inputWords.add(newNumber);
+        }
+        System.out.println(inputWords);
+
+        int firstIndice = scanner.nextInt();
+        int secondIndice = scanner.nextInt();
+        int tmpValue = inputWords.get(secondIndice);
+        inputWords.set(secondIndice, inputWords.get(firstIndice));
+        inputWords.set(firstIndice, tmpValue);
+
+        System.out.println(inputWords);
+    }
+
+    private static void fourteenthExercise() {
+        ArrayList<Character> firstList = new ArrayList<>();
+        firstList.add('a');
+        firstList.add('b');
+        firstList.add('c');
+        firstList.add('d');
+
+        ArrayList<Character> secondList = new ArrayList<>();
+        secondList.add('x');
+        secondList.add('y');
+
+        ArrayList<Character> resultingList = new ArrayList<>();
+
+        for (int i = 0; i < Integer.max(firstList.size(), secondList.size()); i++) {
+            addChar(i, firstList, resultingList);
+            addChar(i, secondList, resultingList);
```

Comment:
Let's remove method addChar() and move its body to here, it will be more readable.
addChar() looks is not good to me, because 1) it has side effect - it changes the value of its parameter. 2) this method is should be called twice to get the needed result with swapped parameters, it is not good if such way will be used in real application, the other developer can use this method only one time and will have a not expected result.

Reply:
fixed